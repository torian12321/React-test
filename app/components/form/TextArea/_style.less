@import '~LESS_conf';

// The css is shared with Input