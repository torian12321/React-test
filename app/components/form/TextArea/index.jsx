var
React     = require('react'),
PropTypes = require('prop-types'),
uniqid    = require('uniqid');

const Textarea = props => {
	const id = props.id || uniqid('textArea_');
	return(
		<textarea
			id           = {id}
			maxLength    = {props.maxlength}
			placeholder  = {props.placeHolder}
			className    = 'form-control'
			onChange     = {props.onChange}
			defaultValue = {props.value}
			disabled     = {props.disabled ? 'disabled' : null}
		/>
	);
}

Textarea.propTypes = {
	id         : PropTypes.string,
	disabled   : PropTypes.bool.isRequired,
	value      : PropTypes.string,
	onChange   : PropTypes.func.isRequired,
	maxlength  : PropTypes.number,
	placeHolder: PropTypes.string
};
Textarea.defaultProps = {
	disabled: false,
	onChange: function(){}
};

module.exports = Textarea;