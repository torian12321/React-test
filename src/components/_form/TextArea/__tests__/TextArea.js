import React    from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../index';

describe('component', () => {
  describe('<TextArea />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <TextArea id='ta' />
      )).toMatchSnapshot();
    });

    it('should render placeholder', () => {
      expect(renderer.create(
        <TextArea id='ta' placeholder='Add your comments' />
      )).toMatchSnapshot();
    });

    it('should render disabled', () => {
      expect(renderer.create(
        <TextArea id='ta' disabled />
      )).toMatchSnapshot();
    });

  });
});