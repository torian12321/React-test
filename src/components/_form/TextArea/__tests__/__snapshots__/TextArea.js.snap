// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <TextArea /> should render correctly 1`] = `
<div
  className="styled-input"
>
  <textarea
    className={null}
    defaultValue={undefined}
    disabled={false}
    id="ta"
    maxLength={undefined}
    onChange={[Function]}
    rows={4}
  />
  <span />
</div>
`;

exports[`component <TextArea /> should render disabled 1`] = `
<div
  className="styled-input"
>
  <textarea
    className={null}
    defaultValue={undefined}
    disabled={true}
    id="ta"
    maxLength={undefined}
    onChange={[Function]}
    rows={4}
  />
  <span />
</div>
`;

exports[`component <TextArea /> should render placeholder 1`] = `
<div
  className="styled-input"
>
  <textarea
    className={null}
    defaultValue={undefined}
    disabled={false}
    id="ta"
    maxLength={undefined}
    onChange={[Function]}
    rows={4}
  />
  <label
    htmlFor="ta"
  >
    Add your comments
  </label>
  <span />
</div>
`;
